### PR TITLE
Refactor unit tests related to hosts and environmental fallbacks

### PR DESCRIPTION
**Pull request by vzhikserg**
Fixes: #628

### CODE REVIEW

File: lib/src/test/java/io/ably/lib/transport/HostsTest.java [hunk 1]
```diff
@@ -69,65 +67,52 @@ public void hosts_fallback_empty_array() throws AblyException {
...
-        }
-        assertThat(idx, is(customHosts.length));
-        assertTrue(allFound);
-        assertTrue(shuffled);
+
+        // Then
+        List<String> fallbackHosts = collectFallbackHosts(hosts);
+        assertThat(hosts.getPrimaryHost(), is(Defaults.HOST_REALTIME));
+        // the returned fallback hosts should have the same elements as custom host fallbacks
+        assertThat(fallbackHosts, containsInAnyOrder(customHosts));
```
Comment (sacOO7) [inline lib/src/test/java/io/ably/lib/transport/HostsTest.java:81]: Nicely implemented 💙

File: lib/src/test/java/io/ably/lib/transport/HostsTest.java [hunk 2]
```diff
@@ -251,21 +233,36 @@ public void hosts_fallback_when_fallback_hosts_and_port_are_defined() throws Abl
...
-        ClientOptions options = new ClientOptions();
+        // Given
         options.tlsPort = 8081;
         options.fallbackHostsUseDefault = true;
+
+        // When
         Hosts hosts = new Hosts(null, Defaults.HOST_REALTIME, options);
 
+        // Then
         assertThat(hosts.getFallback(Defaults.HOST_REALTIME), nullValue());
```
Comment (sacOO7) [inline lib/src/test/java/io/ably/lib/transport/HostsTest.java:257]: It will never reach this statement right

**Comment by sacOO7** [comment]
Btw, I forgot to mention, while naming git branch, we also include issue id, that way it's easy to refer to the exact issue.
Like current branch can be renamed as, ```fix/628-refactor-hosts-unit-tests```
You can start following the convention from next time onwards : )

**Comment by sacOO7** [comment]
Regarding ```private final ClientOptions options = new ClientOptions();```

Just a thought. Normally I would be more careful while using common fields for executing tests. It might create interdependency between tests ( leading to flaky test behavior). **WET** can be preferred over **DRY** in this case : ). But, current tests should work, since it creates ```new HostsTest``` object, every time a new test is executed.

File: lib/src/test/java/io/ably/lib/transport/HostsTest.java (hunk 2 above)
Comment (vzhikserg) [reply lib/src/test/java/io/ably/lib/transport/HostsTest.java:257]: That's right! I will remove this obsolete statement

**Review by sacOO7 — APPROVED** [review]
LGTM 😃

**Review by QuintinWillison — APPROVED** [review]

**Pull request merged**